### PR TITLE
Credits page (close #194)

### LINKED ISSUE #194: Credits page
- New login from the settings page
- list of people in json

properties:
- avatarUrl
- username
- roles

Based on roles, show stacked icons on the right

### PR DESCRIPTION
**Pull request by florian-lefebvre**

### CODE REVIEW

**Comment by florian-lefebvre** [comment]
Fixed :+1

File: contributors.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "description": "God",
+    "bugs": 10,
+    "translation": "fr",
+    "developer": 10,
+    "priority": 0
+  },
+  {
+    "avatarUrl": "https://avatars.githubusercontent.com/u/69633530?v=4",
+    "name": "Florian Lefebvre",
+    "description": "God after God",
```
Comment (skdhg) [inline contributors.json:14]: 😂😂😂

File: src/app/[locale]/(app)/credits/_components/Contributors.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
+"use client";
+
+import Section from "~/components/Section";
+import contributorsData from "../../../../../../contributors.json";
```
Comment (skdhg) [inline src/app/[locale]/(app)/credits/_components/Contributors.tsx:4]: Will it be a good idea to add this to import alias?

File: contributors.json (hunk 1 above)
Comment (florian-lefebvre) [reply contributors.json:14]: I had no inspiration lol

File: src/app/[locale]/(app)/credits/_components/Contributors.tsx (hunk 2 above)
Comment (florian-lefebvre) [reply src/app/[locale]/(app)/credits/_components/Contributors.tsx:4]: I have the same issue with package.json in `src/constants.ts`. Let keep this like this for now but could you open a PR for this?

File: contributors.json [hunk 3]
```diff
@@ -0,0 +1,85 @@
+[
+  {
+    "avatarUrl": "https://avatars.githubusercontent.com/u/42497995?v=4",
+    "name": "@androz2091",
+    "description": "Back-end API developer",
```
Comment (florian-lefebvre) [inline contributors.json:5]: Don't you want to say you're the one in charge / founder / whatever ? @Androz2091

File: contributors.json [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    "bugs": 1000000,
```
Comment (florian-lefebvre) [inline contributors.json:6]: Why so much xd? 10 is enough to show the +9

File: contributors.json [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    "translation": "fr",
+    "developer": 100000,
```
Comment (florian-lefebvre) [inline contributors.json:8]: Same here

File: contributors.json [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    "priority": 0
+  },
+  {
+    "avatarUrl": "https://avatars.githubusercontent.com/u/69633530?v=4",
+    "name": "@florian_lefebvre",
+    "description": "Front-end UI developer",
```
Comment (florian-lefebvre) [inline contributors.json:14]:
```suggestion
    "description": "Front-end developer, UI/UX designer",
```

File: contributors.json [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  {
+    "avatarUrl": "https://cdn.discordapp.com/avatars/876801702051856424/ef0186db796cede85bf0ab34d19d5971.webp?size=80",
+    "name": "@mampo",
+    "bugs": 1
+  },
+  {
+    "avatarUrl": "https://cdn.discordapp.com/avatars/876801702051856424/ef0186db796cede85bf0ab34d19d5971.webp?size=80",
+    "name": "@mampo",
+    "bugs": 1
+  },
```
Comment (florian-lefebvre) [inline contributors.json:64]:
```suggestion
    "avatarUrl": "https://cdn.discordapp.com/avatars/876801702051856424/ef0186db796cede85bf0ab34d19d5971.webp?size=80",
    "name": "@mampo",
    "bugs": 1
  },
```

File: src/app/[locale]/(app)/credits/_components/Contributors.tsx [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  name: string;
+  description?: string;
+  bugs?: number;
+  translation?: string;
+  developer?: number;
+  influencer?: boolean;
+  priority?: number;
+};
+
+// TODO: sort
```
Comment (florian-lefebvre) [inline src/app/[locale]/(app)/credits/_components/Contributors.tsx:26]:
```suggestion
```
Do you think it's working as intended @Androz2091?

File: src/app/[locale]/(app)/credits/_components/Contributors.tsx [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+      title={name}
+      description={
+        description ||
+        `${
+          bugs + developer + (translation ? 1 : 0) + (influencer ? 1 : 0)
+        } contributions`
+      }
+      leftSlot={
+        <div className="relative aspect-square w-10">
+          <DiscordImage
```
Comment (florian-lefebvre) [inline src/app/[locale]/(app)/credits/_components/Contributors.tsx:111]: I'm not confident about this one. Not every contributor will want to put its Discord image. Maybe what we can do is check if the avatarUrl contains `https://discord.com` and load the right component. Tell me if that sounds good to you @Androz2091 and I'll implement

File: src/app/[locale]/(app)/credits/_components/Flag.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import dynamic from "next/dynamic";
+
+export default function Flag({ code }: { code: string }) {
+  const Svg = dynamic(
+    () => import(`flagpack-core/svg/l/${code.toUpperCase()}.svg`),
```
Comment (florian-lefebvre) [inline src/app/[locale]/(app)/credits/_components/Flag.tsx:5]: We'll have to be careful with the translation field in the contributors.json. For instance, `EN` does not exist but `GB-ENG`.
Ref: https://github.com/Yummygum/flagpack-core/tree/main/svg/l

Where to put this link?

File: src/app/[locale]/(app)/settings/_components/About.tsx [hunk 11]
```diff
@@ -55,6 +57,14 @@ export default function About() {
           leftSlot={<LeftIcon icon={SimpleIconsGithub} />}
           rightIcon={ArrowTopRightOnSquareIcon}
         />
+        <DetailCard
+          href="/credits"
+          title="Credits"
+          description="People who helped us"
```
Comment (florian-lefebvre) [inline src/app/[locale]/(app)/settings/_components/About.tsx:63]: We probably need something better here @Androz2091

File: contributors.json [hunk 12]
```diff
@@ -1,20 +1,85 @@
...
+    "name": "@florian_lefebvre",
+    "description": "Front-end UI developer",
     "bugs": 10,
-    "translation": "gb-ukm",
+    "translation": "fr",
     "developer": 10,
     "priority": 1
+  },
+  {
+    "avatarUrl": "https://cdn.discordapp.com/avatars/236952680201715714/9f8f56730a097338968b19016b43e58c.webp?size=40",
```
Comment (skdhg) [inline contributors.json:21]: hardcoding discord avatar url might not be a good idea 🤔

Comment (skdhg) [reply contributors.json:21]: github avatars can be github.com/username.png which will always exist but this is not the case for discord avatars

File: src/app/[locale]/(app)/credits/_components/Contributors.tsx (hunk 9 above)
Comment (skdhg) [reply src/app/[locale]/(app)/credits/_components/Contributors.tsx:111]: I guess we can turn off conditional rendering inside `<DiscordImage/>` if the host does not match `discord.com`/`discordapp.com`, does that sound good?

File: contributors.json (hunk 3 above)
Comment (Androz2091) [reply contributors.json:5]: Not really. People can use contact@dumpus.app to contact us

File: src/app/[locale]/(app)/credits/_components/Contributors.tsx (hunk 9 above)
Comment (Androz2091) [reply src/app/[locale]/(app)/credits/_components/Contributors.tsx:111]: Yes @skdhg, that would be perfect

File: contributors.json (hunk 12 above)
Comment (Androz2091) [reply contributors.json:21]: Yes @skdhg indeed. We'll host a low quality avatar in this repository

File: src/app/[locale]/(app)/credits/_components/Contributors.tsx (hunk 9 above)
Comment (florian-lefebvre) [reply src/app/[locale]/(app)/credits/_components/Contributors.tsx:111]: Do you want to make a PR for it @skdhg?

**Pull request merged**